src/main/scala/ai/privado/exporter/ExporterUtility.scala

```diff
@@ -193,9 +200,14 @@ object ExporterUtility {
     index: Int = -1,
     sizeOfList: Int = -1,
     messageInExcerpt: String = "",
-    appCache: AppCache
+    appCache: AppCache,
+    ruleCache: RuleCache
   ): Option[DataFlowSubCategoryPathExcerptModel] = {
-    val sample = node.code
+    val allowedLineLimit = ruleCache.getSystemConfigByKey(Constants.MaxLineLimit, true)
+    val limitValue       = if (allowedLineLimit.nonEmpty) allowedLineLimit.toInt else 0
+    val sample =
+      if (limitValue != 0 && node.code.length > limitValue) node.code.take(limitValue) + "..." else node.code
```

Comment:
This piece of code can be written as 
```
 val sample = ruleCache.getSystemConfigByKey(Constants.MaxLineLimit, true).flatMap(_.toIntOption) match {
    case Some(limit) if (limit > 3 && node.code.length > limit) => node.code.take(limit-3) + "..."
    case _ => node.code
  }
  ```
  
  we minus `3` because of addition of `...` in code

Reply:
Also I see you have applied the limit on `sample`, do we need to apply on `sample` or `excerpt`?

Reply:
Yes because the long string present in `sample` or `excerpt` can hang the browser to stuck while rendering. This two parameter contains the string that hold lengthy string.